src/v0/destinations/marketo/util.js

```diff
@@ -122,17 +123,27 @@ const marketoApplicationErrorHandler = (marketoResponse, sourceMessage, destinat
 const nestedResponseHandler = (marketoResponse, sourceMessage) => {
   const checkStatus = (res) => {
     const { status } = res;
-    if (status && status !== 'updated' && status !== 'created' && status !== 'added') {
+    const allowedStatus = ['updated', 'added', 'removed', 'created'];
+    if (
+      status &&
+      !allowedStatus.includes(status)
+      // we need to check the case where the id are not in list
+    ) {
       const { reasons } = res;
       let statusCode = 400;
-      if (reasons && RECORD_LEVEL_ABORTBALE_ERRORS.includes(reasons[0].code)) {
-        statusCode = 400;
-      } else if (reasons && MARKETO_ABORTABLE_CODES.includes(reasons[0].code)) {
-        statusCode = 400;
-      } else if (reasons && MARKETO_THROTTLED_CODES.includes(reasons[0].code)) {
-        statusCode = 429;
-      } else if (reasons && MARKETO_RETRYABLE_CODES.includes(reasons[0].code)) {
-        statusCode = 500;
+      if (reasons) {
+        const errorCodesFromDest = reasons.map((reason) => reason.code);
+        const filteredErrorCode = errorCodesFromDest.find(
+          (errorCode) => !ERROR_CODE_TO_PASS.includes(errorCode),
+        );
+        if (!filteredErrorCode) {
+          return;
+        }
+        if (MARKETO_THROTTLED_CODES.includes(filteredErrorCode)) {
+          statusCode = 429;
+        } else if (MARKETO_RETRYABLE_CODES.includes(filteredErrorCode)) {
+          statusCode = 500;
+        }
```

Comment:
throw new InstrumentationError(
        `Request failed during: ${sourceMessage}, error: ${JSON.stringify(reasons)}`, <-- lets add some default error message here as well

Reply:
Also, why 601 is retryable here. In marketo bulk upload we marked it abortable

Reply:
What do we mean by default error message ?
Source Message is already there and due its complexity we can't say anything by default

Reply:
Added a default message of Reasons not found

Reply:
Also what is your take on the invalid access token status code?

Reply:
It should result it in retryable error as it will generate a new access token and perform the operation

Reply:
as per the offline chat, @anantjain45823 confirmed that invalid credentials does not return 601 status code.